_docs/_latest/develop/dev/gcp-api.md

```diff
@@ -0,0 +1,220 @@
+        """
+
+        paged_results = self.repository.instances.list(project_id)
+        flattened_results = api_helpers.flatten_list_results(
+            paged_results, 'items')
+        LOGGER.debug('Getting all the cloudsql instances of a project,'
+                     ' project_id = %s, flattened_results = %s',
+                      project_id, flattened_results)
+        return flattened_results
+```
+
+## Create FooRepositoryClient class, and define the property method
+
+1. Create a FooRepositoryClient class in foo.py.
+1. Inherit from the `BaseRepositoryClient`.
+1. Initialize it with quota parameters.
+1. Define the property method named after the bar resource we want to interact
+with, e.g. bar(). This property method will initialize a \_FooBarRepository
+class (see next section), which will install the base GCP API functionalities
+(building requests, authentication, and API methods).
+
+
+[google/cloud/forseti/common/gcp_api/cloudsql.py]({% link _docs/_latest/develop/reference/_modules/google/cloud/forseti/common/gcp_api/cloudsql.html %})
+
+In this example, bar is ```instances```.
+
+```python
+class CloudSqlRepositoryClient(_base_repository.BaseRepositoryClient):
+    """Cloud SQL Admin API Respository."""
+
+    def __init__(self,
+                 quota_max_calls=None,
+                 quota_period=1.0,
+                 use_rate_limiter=True):
+        """Constructor.
+
+        Args:
+            quota_max_calls (int): Allowed requests per <quota_period> for the
+                API.
+            quota_period (float): The time period to track requests over.
+            use_rate_limiter (bool): Set to false to disable the use of a rate
+                limiter for this service.
+        """
+        if not quota_max_calls:
+            use_rate_limiter = False
+
+        self._instances = None
+
+        super(CloudSqlRepositoryClient, self).__init__(
+            'sqladmin', versions=['v1beta4'],
+            quota_max_calls=quota_max_calls,
+            quota_period=quota_period,
+            use_rate_limiter=use_rate_limiter)
+
+    @property
+    def instances(self):
+        """Returns a _CloudSqlInstancesRepository instance."""
+        if not self._instances:
+            self._instances = self._init_repository(
+                _CloudSqlInstancesRepository)
+        return self._instances
+```
+
+## Create \_FooBarRepository class, to install the base GCP API functionalities
+
+1. Create a \_FooBarRepository class in foo.py.
+1. Make it inherit the base `GCPRepository` and the appropriate GCP API method
+mixins.  In this example, the mixin for list.
+1. This will install the base GCP API functionalities for building requests,
+authentication, and the API methods.
+
+[google/cloud/forseti/common/gcp_api/cloudsql.py]({% link _docs/_latest/develop/reference/_modules/google/cloud/forseti/common/gcp_api/cloudsql.html %})
+
+```python
+class _CloudSqlInstancesRepository(
```

Comment:
If we can get away without copying in code here, we're better off, maybe just link to the generated source in /reference?